app/lib/task/scheduler.dart

```diff
@@ -307,14 +308,38 @@ Future<Payload?> updatePackageStateWithPendingVersions(
     s.derivePendingAt();
     tx.insert(s);
 
+    final payloadVersions = pendingVersions
+        .map((v) => VersionTokenPair(
+              version: v,
+              token: s.versions![v]!.secretToken!,
+            ))
+        .toList();
+
+    // sort pending versions based on version priority
+    if (payloadVersions.length > 1) {
+      // loading the package entity to prioritize latest versions
+      final pkg = await tx.lookupOrNull<Package>(
+          db.emptyKey.append(Package, id: state.package));
```

Comment:
How about no using this? just using the version number.

I/O is not super cheap, and we need to schedule many instances quickly.